### ISSUE TITLE
[PRE REVIEW]: COSplay: Contrast Optimized Stimulation Player

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1039]
**Submitting author:** @TheChymera (<a href="http://orcid.org/0000-0001-7037-2449">Horea-Ioan Ioanas</a>)
**Repository:** <a href="https://github.com/IBT-FMI/COSplay/" target ="_blank">https://github.com/IBT-FMI/COSplay/</a>
**Version:** 1.1.1
**Editor:** @arokem
**Reviewers:** damaggu

**Author instructions**

Thanks for submitting your paper to JOSS @TheChymera. The JOSS editor (shown at the top of this issue) will work with you on this issue to find a reviewer for your submission before creating the main review issue.

@TheChymera if you have any suggestions for potential reviewers then please mention them here in this thread. In addition, [this list of people](https://bit.ly/joss-reviewers) have already agreed to review for JOSS and may be suitable for this submission.

**Editor instructions**

The JOSS submission bot @whedon is here to help you find and assign reviewers and start the main review. To find out what @whedon can do for you type:

```
@whedon commands
```

**Comment by arfon** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01039/joss.01039/10.21105.joss.01039.pdf)

**Comment by arfon** [comment]
👋 @arokem - would you be willing to edit this submission for JOSS?

**Comment by labarba** [comment]
@TheChymera : I had a quick look at your submission, and I'm a little concerned about how a reviewer might approach this. The software is meant to be used with a device, the **COSplayer**, which is custom-built open-source hardware using a PyBoard. Do you provide any way for the software to be tested _without_ this device? It would be useful, for example, to have synthetic data and a toy workflow that could be executed to demo the functionality, without the hardware. Is that possible?

**Comment by TheChymera** [comment]
ugh, not really, the software interacts with the hardware. At the very least you would need a pyboard, and then you could observe the LED pattern.

If even that is unavailable you would need a software emulator for the pyboard, which is very much nontrivial to produce and would add little value to the project (others have attempted to make such emulators and given up). Unless such an emulator is guaranteed to be perfect (which is impossible), errors appearing in this use case might as well be errors with the emulator.

What I could offer, if a pyboard truly is unavailable, is to mail a pyboard to the reviewer (which we would like to have back :) ).

We had hoped that an interested reviewer might actually attempt to reproduce the device as per our complete video guide [[1](https://www.youtube.com/watch?v=YakrW-PoAgM),[2](https://figshare.com/articles/A_Guide_to_Assembling_the_COSplayer_an_Open_Source_Device_for_Microsecond-Range_Stimulus_Delivery_with_broad_Application_in_Biomedical_Engineering_and_fMRI/7227626)]. The best part about this is that the reviewer would then get to keep and use the COSplayer, which is a really cool device!

**Comment by danielskatz** [comment]
👋 @arokem - would you be willing to edit this submission for JOSS? (reposting)

**Comment by danielskatz** [comment]
👋 @arokem - ping...

**Comment by arokem** [comment]
So sorry for missing this! I will take this on.

**Comment by arokem** [comment]
@whedon assign @arokem as editor

**Comment by whedon** [comment]
OK, the editor is @arokem

**Comment by arokem** [comment]
@TheChymera : so sorry for the delay here.

I was thinking of asking folks in the PsychoPy developer community to review this. Do you have any other ideas for anyone who might be interested in reviewing this?

**Comment by TheChymera** [comment]
I agree that there might be some overlap in the focus of these projects. Psychopy is more specifically targeted to high-level human applications and complex stimulus generation, whereas COSplay is mainly targeted at animal applications and easy integration with animal imaging workflows.

As this community is smaller and more tightly knit, I am afraid I could only recommend reviewers who actually work with us. However there might be people in the broader Python neuroimaging community who could offer valuable input not biased by prior exposure to the project. Perhaps @chrisfilo , @satra , @oesteban , @effigies , or @salma1601 ? But sadly I cannot comment on their willingness to review.

**Comment by chrisfilo** [comment]
Sorry - snowed under.

**Comment by oesteban** [comment]
My apologies, I'm up to the neck too.

**Comment by arokem** [comment]
@satra: are you possibly available to review this paper for JOSS?

**Comment by satra** [comment]
@arokem - sorry not in the next one month.

**Comment by arokem** [comment]
@effigies : are you available to review this paper for JOSS?

**Comment by TheChymera** [comment]
@damaggu could help out. He also has access to a Bruker scanner similar to ours. I think he has already crossed paths with one of the copies of the device floating around campus, but he is not a contributor.

**Comment by damaggu** [comment]
@effigies @TheChymera Yeah, I'd be happy to help.

**Comment by arokem** [comment]
I think that reproducing the device is too tall an order for a reviewer on the paper. I think that checking that everything matches the requirements and offering feedback could be done even without direct access to such a device. I suggest that @effigies and @damaggu both conduct the review and that you use the repo issues as a public place to discuss any issues between you. 

Thanks everyone!

**Comment by arokem** [comment]
@whedon assign effigies as reviewer

**Comment by whedon** [comment]
OK, the reviewer is effigies

**Comment by arokem** [comment]
@whedon assign damaggu as reviewer

**Comment by arokem** [comment]
@whedon start review

**Comment by whedon** [comment]
OK, the reviewer is damaggu

**Comment by whedon** [comment]
OK, I've started the review over in https://github.com/openjournals/joss-reviews/issues/1171. Feel free to close this issue now!

**Comment by arokem** [comment]
I'll close this one. Please head over to #1171 for the review.

**Comment by arfon** [comment]
Just a heads up @arokem - using `assign` clobbers existing reviewer assignments. You need to use `add` to add additional reviewers when some already exist. I've fixed up the review issue over in #1171.
